arctic/chunkstore/chunkstore.py

```diff
@@ -0,0 +1,376 @@
+
+    def read(self, symbol, chunk_range=None):
+        """
+        Reads data for a given symbol from the database.
+
+        Parameters
+        ----------
+        symbol: str
+            the symbol to retrieve
+        chunk_range: object
+            corresponding range object for the specified chunker (for
+            DateChunker it is a DateRange object)
+
+        Returns
+        -------
+        A dataframe or series
+        """
+
+        sym = self._get_symbol_info(symbol)
+
+        if not sym:
+            raise NoDataFoundException('No data found for %s in library %s' % (symbol, self._collection.get_name()))
+
+        spec = {'symbol': symbol,
+                'parent': sym['_id'],
+                }
+
+        if chunk_range is not None:
+            spec['start'] = self.chunker.to_mongo(chunk_range)
+
+        segments = []
+        i = -1
+        for i, x in enumerate(self._collection.find(spec, sort=[('start', pymongo.ASCENDING)],)):
+            segments.append(decompress(x['data']))
+
+        data = b''.join(segments)
+
+        dtype = PandasSerializer()._dtype(sym['dtype'], sym.get('dtype_metadata', {}))
+        records = np.fromstring(data, dtype=dtype).reshape(sym.get('shape', (-1)))
+
+        data = deserialize(records, sym['type'])
+
+        if chunk_range is None:
+            return data
+        return self.chunker.filter(data, chunk_range)
+
+    def write(self, symbol, item, chunk_size):
+        """
+        Writes data from item to symbol in the database
+
+        Parameters
+        ----------
+        symbol: str
+            the symbol that will be used to reference the written data
+        item: dataframe or series
+            the data to write the database
+        chunk_size: ?
+            A chunk size that is understood by the specified chunker
+        """
+
+        doc = {'_id': bson.ObjectId()}
+        doc['symbol'] = symbol
+        doc['chunk_size'] = chunk_size
+
+        if isinstance(item, Series):
+            doc['type'] = SeriesSerializer.TYPE
+        elif isinstance(item, DataFrame):
+            doc['type'] = DataFrameSerializer.TYPE
+        else:
+            raise Exception("Can only chunk Series and DataFrames")
+
+        records = []
+        ranges = []
+        dtype = None
+
+        for start, end, record in self.chunker.to_chunks(item, chunk_size):
+            r, dtype = serialize(record, string_max_len=self.STRING_MAX)
+            records.append(r)
+            ranges.append((start, end))
+
+        item = np.array([r for record in records for r in record]).flatten()
+        for record in records:
+            if record.dtype.hasobject:
+                raise UnhandledDtypeException()
+
+        sym = self._get_symbol_info(symbol)
+        if sym:
+            # if the symbol already exists, we are basically overwriting it
+            # clean up the data before we orphan the symbol chunks
+            self.delete(symbol)
```

Comment:
Quite often people overwrite data - esp. during development. Can you just delete the old chunks once the new writes have gone through?